src/components/ClipboardAwarePasteButton.tsx

```diff
@@ -30,14 +39,54 @@ export default function ClipboardAwarePasteButton({
     return null
   }
 
+  // Debounce onPress event so that it is called once on trigger and
+  // consecutive calls in given period are ignored.
+  const debouncedOnPress = debounce(
+    async () => {
+      await onPressInternal()
+    },
+    BUTTON_TAP_DEBOUNCE_TIME,
+    DEBOUNCE_OPTIONS
+  )
+
   return (
-    <Button
-      text={t('paste')}
-      type={BtnTypes.ONBOARDING_SECONDARY}
-      rounded={false}
-      size={BtnSizes.FULL}
-      onPress={onPressInternal}
-      testID={'PasteButton'}
-    />
+    <View style={styles.container}>
+      {/* these Views cannot be combined as it will cause ripple to not respect the border radius */}
+      <View style={styles.containRipple}>
+        <Touchable onPress={debouncedOnPress} style={styles.button} testID={'PasteButton'}>
+          <>
+            <Text maxFontSizeMultiplier={1} style={styles.fontStyle}>
+              {t('paste')}
+            </Text>
+          </>
+        </Touchable>
+      </View>
+    </View>
   )
-}
+})
+
+const styles = StyleSheet.create({
+  // on android Touchable Provides a ripple effect, by itself it does not respect the border radius on Touchable
+  containRipple: {
+    overflow: 'hidden',
+  },
+  container: {
+    flexDirection: 'column',
+  },
+  button: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingVertical: 5,
+    paddingHorizontal: 24,
```

Comment:
nit: use spacing constants for padding / margin where possible throughout the PR